elastic/integrations
# [Carbon Black Cloud] - Added alerts v7 data stream and CEL input type

## Type of change
- Enhancement

## Proposed commit message
Added a new alerts_v7 data stream to mirror the v7 api and schema changes from carbon black cloud.
The old data stream has not been removed in order to allow smooth transition to the new v7 data stream once the old api stops working. This avoids a breaking change. In addition to this a new CEL input option has been introduced to bring more flexibility for the future. This new input has been tagged as [Beta] while the existing HTTPJSON input has been tagged as [Legacy]. This was done in order to smoothen the transition overtime and avoid breaking changes.

## Checklist

## Author's Checklist
1. New alert v7 data stream added with tests.
2. Alert v7 Dashboards & Search added.
3. Documentation updated.
4. Added new [Beta] CEL input type for HTTP api dependant data streams. (alert_v7, asset_vulnerability_summary, audit)

## Note
1. The existing HTTPJSON input is being marked as [Legacy] with the introduction of the CEL input. 
2. Please ignore commits before March 26th. Start from ([initial commit, new v7 data stream under construction (ignore for review)](https://github.com/elastic/integrations/pull/9467/commits/bf8ecbd77184e8882d0b6dafd68ced1217a2818d)


## How to test this PR locally

Clone integrations repo.
Install the elastic package locally.
Start the elastic stack using the **elastic package stack up** command.
Move to integrations/packages/carbon_black_cloud directory.
Run the following command to run tests.
1. **elastic-package build && elastic-package format && elastic-package lint && elastic-package check && elastic-package build**
2. **elastic-package stack up -v -d  --services package-registry**
3. Bind the shell using the command - **eval "$(elastic-package stack shellinit)"** or **eval $(elastic-package stack shellinit --shell bash)** on mac m1
4. Use  the command - **elastic-package test -v** to test


## Related issues
- Closes #7997
- Related https://github.com/elastic/enhancements/issues/21160

## Test Results
```yml
--- Test results for package: carbon_black_cloud - START ---
╭────────────────────┬─────────────────────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE            │ DATA STREAM                 │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├────────────────────┼─────────────────────────────┼───────────┼───────────┼────────┼───────────────┤
│ carbon_black_cloud │ alert                       │ system    │ default   │ PASS   │    32.715904s │
│ carbon_black_cloud │ alert_v7                    │ system    │ httpjson  │ PASS   │ 30.962756791s │
│ carbon_black_cloud │ alert_v7                    │ system    │ cel       │ PASS   │ 32.664833208s │
│ carbon_black_cloud │ asset_vulnerability_summary │ system    │ httpjson  │ PASS   │ 28.750989958s │
│ carbon_black_cloud │ asset_vulnerability_summary │ system    │ cel       │ PASS   │ 31.468396708s │
│ carbon_black_cloud │ audit                       │ system    │ httpjson  │ PASS   │ 32.579871041s │
│ carbon_black_cloud │ audit                       │ system    │ cel       │ PASS   │ 31.452701917s │
╰────────────────────┴─────────────────────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: carbon_black_cloud - END   ---
```

## Screenshots

Add here screenshots presenting:
- Kibana UI forms presenting configuration options exposed by the integration:
   
<img width="1427" alt="Screenshot 2024-04-05 at 9 10 58 PM" src="https://github.com/elastic/integrations/assets/105607378/59b052c8-c243-4cba-8fa8-172fef27d0a6">

- Dashboards with collected metrics or logs:
   
   **Alerts V7 (Samples):** 
   
<img width="1721" alt="alerts_v7_1" src="https://github.com/elastic/integrations/assets/105607378/720b56fd-c01c-4789-a34b-15b47e51de34">

<img width="1721" alt="alerts_v7_2" src="https://github.com/elastic/integrations/assets/105607378/dfce9eeb-0e55-4b72-b5a8-db917a5448ba">

<img width="1721" alt="alerts_v7_3" src="https://github.com/elastic/integrations/assets/105607378/aeb45e3f-b9f1-4fcf-a47a-a34481c82514">

<img width="1721" alt="alerts_v7_4" src="https://github.com/elastic/integrations/assets/105607378/2c898648-efe0-4306-aea6-04de034d5b61">

**Asset Vulnerability:**

<img width="1721" alt="asset_vulnerability" src="https://github.com/elastic/integrations/assets/105607378/d13bf601-c052-421d-a5c5-009bdcb5d077">

**Audit Logs:**

<img width="1721" alt="audit_logs" src="https://github.com/elastic/integrations/assets/105607378/ec6de91e-68d1-4f4f-8b81-abff6eae374d">
